### DISCUSSION TITLE
Password is sent in clear text in login page

### DISCUSSION THREAD

**Discussion by ramup1717** [keycloak/keycloak#14876] [Q&A] [answered]
Dear Team, 
 In one of our project we are using keycloak18 for user authentication. we have submitted our application for VAPT. audit team raised a concern on password is in clear text and it should be encrypted. So kindly guide us how to resolve this issue. Even we tried solution mentioned in "https://github.com/edwin/keycloak-password-encryptor" url.  we are unable to see the Custom SPI in admin authentication workflow after keeping custom Jar in deployments folder. kindly help us in this regard.

Thanks

**Comment by sschu** [comment, marked as answer]
The password is stored hashed and salted by default. If you have enabled TLS (which is required for production use), the password is also transferred encrypted. Keycloak TLS setup is explained here: https://www.keycloak.org/docs/latest/server_installation/#_setting_up_ssl

**Reply by ramup1717** [reply]
thank you.
